app/Services/CampaignService.php

```diff
@@ -176,4 +177,33 @@ public function getCampaignIdsFromSignups()
 
         return $ids ? $ids : null;
     }
+
+    /**
+     * Appends status counts to a collection of campaigns.
+     *
+     * @return Illuminate\Database\Eloquent\Collection $campaigns
+     */
+    public function getCampaignPostStatusCounts($campaigns)
+    {
+        $ids = array_pluck($campaigns, 'id');
+
+        $campaignsWithSignups = Signup::campaign($ids)
+            ->includePostStatusCounts()
+            ->get()
+            ->groupBy('campaign_id');
```

Comment:
@DFurnes actually this isn't going to work. I need all of the individual signups with their post counts keyed by campaign_id so that I can then sum them. If I try to do the `groupBy` before the `get` I only get one signup per campaign_id. 

If it is more performant to do this in sql then I could try to write a sql query that will get me campaigns and counts without the collection stuff. Do you think that would be better.